src/server/mod.rs

```diff
@@ -29,19 +30,33 @@ const REQWEST_USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARG
 
 /// This is the global HTTP state object that will be made available to all
 /// HTTP API calls.
-#[derive(Clone)]
 pub struct ServerState {
     /// Metric reporting
     pub metrics: Box<StatsdClient>,
     pub adm_endpoint_url: String,
     pub reqwest_client: reqwest::Client,
     pub tiles_cache: cache::TilesCache,
     pub settings: Settings,
-    pub filter: AdmFilter,
+    pub filter: RwLock<AdmFilter>,
     pub img_store: Option<StoreImage>,
     pub excluded_dmas: Option<Vec<u16>>,
 }
 
+impl Clone for ServerState {
+    fn clone(&self) -> Self {
+        Self {
+            metrics: self.metrics.clone(),
+            adm_endpoint_url: self.adm_endpoint_url.clone(),
+            reqwest_client: self.reqwest_client.clone(),
+            tiles_cache: self.tiles_cache.clone(),
+            settings: self.settings.clone(),
+            filter: RwLock::new((*self.filter.read().unwrap()).clone()),
```

Comment:
We don't want to clone the `AdmFilter` here, ending up with separate copies because the background updater only updates the original copy. I think we can wrap the `RwLock` in an `Arc` instead.